python/pyarrow/types.py

```diff
@@ -64,17 +82,17 @@ def is_boolean(t):
 
 @doc(is_null, datatype="any integer")
 def is_integer(t):
-    return t.id in _INTEGER_TYPES
+    return is_integer_type(t)
```

Comment:
Hmm... why are we defining two sets of functions exactly? We should have either `is_integer` or `is_integer_type`, not both.

Reply:
hello. I would appreciate it if you could review it when you have time. thank you! cc @pitrou